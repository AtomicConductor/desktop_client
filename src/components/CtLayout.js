import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import CssBaseline from "@material-ui/core/CssBaseline";
import CtDrawer from "./CtDrawer";
import CtDownloader from "./downloader/CtDownloader";
import CtUploader from "./uploader/CtUploader";
import CtPlugins from "./plugins/CtPlugins";
import CtDashboard from "./dashboard/CtDashboard";
import {
  BrowserRouter as Router,
  Route,
  Switch,
  Redirect
} from "react-router-dom";

const useStyles = makeStyles(theme => ({
  root: {
    display: "flex"
  }
}));

function CtLayout() {
  const classes = useStyles();

  return (
    <Router>
      <div className={classes.root}>
        <CssBaseline />
        <CtDrawer />
        <Switch>
          <Route
            path="/dashboard"
            render={props => <CtDashboard {...props} />}
          />
          <Route
            path="/downloader"
            render={props => <CtDownloader {...props} />}
          />
          <Route
            path="/uploader"
            render={props => <CtUploader {...props} />}
          />
          <Route
            path="/plugins"
            render={props => <CtPlugins {...props} />}
          />
          <Redirect push to="/dashboard" />
        </Switch>
      </div>
    </Router>
  );
}

export default CtLayout;
