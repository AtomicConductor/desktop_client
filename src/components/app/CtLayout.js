/* 
Manage high level layout of page.

All top level components exist here.
Main box

Drawer
Status line
Snackbars
Modals

The main box also contains the router.
Other elements live outside the router. 
*/

import React, { useEffect } from "react";

import { makeStyles } from "@material-ui/core/styles";
import Box from "@material-ui/core/Box";

import CtDrawerContainer from "./CtDrawerContainer";
import Downloader from "../downloader/Downloader";
import CtUploader from "../uploader/CtUploader";
import CtPlugins from "../plugins/CtPlugins";
import CtAccountContainer from "../account/CtAccountContainer";

import CtDashboard from "../dashboard/CtDashboard";
import CtSettings from "../settings/CtSettings";
import StatusLine from "./StatusLine";

import CtSnackbarContainer from "../notification/CtSnackbarContainer";
import { drawerWidth, statusLineHeight } from "../../_helpers/constants";

import {
  BrowserRouter as Router,
  Route,
  Switch,
  Redirect
} from "react-router-dom";

const useStyles = makeStyles(theme => ({
  box: {
    // border: "1px solid #0f0",
    height: window.innerHeight - statusLineHeight,
    marginLeft: drawerWidth
  }
}));

const CtLayout = () => {
  const classes = useStyles();

  return (
    <Router>
      <CtDrawerContainer />
      <Box className={classes.box}>
        <Switch>
          <Route
            path="/dashboard"
            render={props => <CtDashboard {...props} />}
          />
          <Route
            path="/downloader/jobs"
            render={props => <Downloader {...props} />}
          />
          <Route
            path="/downloader/queue"
            render={props => <Downloader {...props} />}
          />
          <Route
            path="/downloader"
            render={props => <Downloader {...props} />}
          />
          <Route path="/uploader" render={props => <CtUploader {...props} />} />
          <Route path="/plugins" render={props => <CtPlugins {...props} />} />
          <Route path="/settings" render={props => <CtSettings {...props} />} />
          <Route
            path="/account"
            render={props => <CtAccountContainer {...props} />}
          />

          <Redirect push to="/dashboard" />
        </Switch>
      </Box>
      <CtSnackbarContainer />
      <StatusLine />
    </Router>
  );
};

export default CtLayout;
