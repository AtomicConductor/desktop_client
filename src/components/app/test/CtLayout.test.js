import React from "react";
import { shallow } from "enzyme";
import CtLayout from "../CtLayout";

const setup = overrides => {
  const props = { ...overrides };
  const wrapper = shallow(<CtLayout {...props} />);
  return {
    props,
    wrapper
  };
};

it("renders without crashing", () => {
  const wrapper = setup().wrapper;
  expect(wrapper.exists()).toBe(true);
});

describe("snapshots", () => {
  it("match snapshot", () => {
    const wrapper = setup().wrapper;
    expect(wrapper).toMatchSnapshot();
  });
});
