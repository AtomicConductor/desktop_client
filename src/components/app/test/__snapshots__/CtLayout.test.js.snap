// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CtLayout />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <div
          className="makeStyles-root-1"
        >
          <CssBaseline />
          <UNDEFINED />
          <Switch>
            <Route
              path="/dashboard"
              render={[Function]}
            />
            <Route
              path="/downloader"
              render={[Function]}
            />
            <Route
              path="/uploader"
              render={[Function]}
            />
            <Route
              path="/plugins"
              render={[Function]}
            />
            <Route
              path="/settings"
              render={[Function]}
            />
            <Route
              path="/account"
              render={[Function]}
            />
            <Redirect
              push={true}
              to="/dashboard"
            />
          </Switch>
        </div>,
        <UNDEFINED />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <CssBaseline />,
            <UNDEFINED />,
            <Switch>
              <Route
                path="/dashboard"
                render={[Function]}
              />
              <Route
                path="/downloader"
                render={[Function]}
              />
              <Route
                path="/uploader"
                render={[Function]}
              />
              <Route
                path="/plugins"
                render={[Function]}
              />
              <Route
                path="/settings"
                render={[Function]}
              />
              <Route
                path="/account"
                render={[Function]}
              />
              <Redirect
                push={true}
                to="/dashboard"
              />
            </Switch>,
          ],
          "className": "makeStyles-root-1",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(CtDrawer)",
              "type": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  path="/dashboard"
                  render={[Function]}
                />,
                <Route
                  path="/downloader"
                  render={[Function]}
                />,
                <Route
                  path="/uploader"
                  render={[Function]}
                />,
                <Route
                  path="/plugins"
                  render={[Function]}
                />,
                <Route
                  path="/settings"
                  render={[Function]}
                />,
                <Route
                  path="/account"
                  render={[Function]}
                />,
                <Redirect
                  push={true}
                  to="/dashboard"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "path": "/dashboard",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "path": "/downloader",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "path": "/uploader",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "path": "/plugins",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "path": "/settings",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "path": "/account",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "push": true,
                  "to": "/dashboard",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(CtSnackbar)",
          "type": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div
            className="makeStyles-root-1"
          >
            <CssBaseline />
            <UNDEFINED />
            <Switch>
              <Route
                path="/dashboard"
                render={[Function]}
              />
              <Route
                path="/downloader"
                render={[Function]}
              />
              <Route
                path="/uploader"
                render={[Function]}
              />
              <Route
                path="/plugins"
                render={[Function]}
              />
              <Route
                path="/settings"
                render={[Function]}
              />
              <Route
                path="/account"
                render={[Function]}
              />
              <Redirect
                push={true}
                to="/dashboard"
              />
            </Switch>
          </div>,
          <UNDEFINED />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <CssBaseline />,
              <UNDEFINED />,
              <Switch>
                <Route
                  path="/dashboard"
                  render={[Function]}
                />
                <Route
                  path="/downloader"
                  render={[Function]}
                />
                <Route
                  path="/uploader"
                  render={[Function]}
                />
                <Route
                  path="/plugins"
                  render={[Function]}
                />
                <Route
                  path="/settings"
                  render={[Function]}
                />
                <Route
                  path="/account"
                  render={[Function]}
                />
                <Redirect
                  push={true}
                  to="/dashboard"
                />
              </Switch>,
            ],
            "className": "makeStyles-root-1",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(CtDrawer)",
                "type": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Route
                    path="/dashboard"
                    render={[Function]}
                  />,
                  <Route
                    path="/downloader"
                    render={[Function]}
                  />,
                  <Route
                    path="/uploader"
                    render={[Function]}
                  />,
                  <Route
                    path="/plugins"
                    render={[Function]}
                  />,
                  <Route
                    path="/settings"
                    render={[Function]}
                  />,
                  <Route
                    path="/account"
                    render={[Function]}
                  />,
                  <Redirect
                    push={true}
                    to="/dashboard"
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "path": "/dashboard",
                    "render": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "path": "/downloader",
                    "render": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "path": "/uploader",
                    "render": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "path": "/plugins",
                    "render": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "path": "/settings",
                    "render": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "path": "/account",
                    "render": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "push": true,
                    "to": "/dashboard",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(CtSnackbar)",
            "type": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`snapshots match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CtLayout />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <div
          className="makeStyles-root-1"
        >
          <CssBaseline />
          <UNDEFINED />
          <Switch>
            <Route
              path="/dashboard"
              render={[Function]}
            />
            <Route
              path="/downloader"
              render={[Function]}
            />
            <Route
              path="/uploader"
              render={[Function]}
            />
            <Route
              path="/plugins"
              render={[Function]}
            />
            <Route
              path="/settings"
              render={[Function]}
            />
            <Route
              path="/account"
              render={[Function]}
            />
            <Redirect
              push={true}
              to="/dashboard"
            />
          </Switch>
        </div>,
        <UNDEFINED />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <CssBaseline />,
            <UNDEFINED />,
            <Switch>
              <Route
                path="/dashboard"
                render={[Function]}
              />
              <Route
                path="/downloader"
                render={[Function]}
              />
              <Route
                path="/uploader"
                render={[Function]}
              />
              <Route
                path="/plugins"
                render={[Function]}
              />
              <Route
                path="/settings"
                render={[Function]}
              />
              <Route
                path="/account"
                render={[Function]}
              />
              <Redirect
                push={true}
                to="/dashboard"
              />
            </Switch>,
          ],
          "className": "makeStyles-root-1",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(CtDrawer)",
              "type": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  path="/dashboard"
                  render={[Function]}
                />,
                <Route
                  path="/downloader"
                  render={[Function]}
                />,
                <Route
                  path="/uploader"
                  render={[Function]}
                />,
                <Route
                  path="/plugins"
                  render={[Function]}
                />,
                <Route
                  path="/settings"
                  render={[Function]}
                />,
                <Route
                  path="/account"
                  render={[Function]}
                />,
                <Redirect
                  push={true}
                  to="/dashboard"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "path": "/dashboard",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "path": "/downloader",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "path": "/uploader",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "path": "/plugins",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "path": "/settings",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "path": "/account",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "push": true,
                  "to": "/dashboard",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(CtSnackbar)",
          "type": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div
            className="makeStyles-root-1"
          >
            <CssBaseline />
            <UNDEFINED />
            <Switch>
              <Route
                path="/dashboard"
                render={[Function]}
              />
              <Route
                path="/downloader"
                render={[Function]}
              />
              <Route
                path="/uploader"
                render={[Function]}
              />
              <Route
                path="/plugins"
                render={[Function]}
              />
              <Route
                path="/settings"
                render={[Function]}
              />
              <Route
                path="/account"
                render={[Function]}
              />
              <Redirect
                push={true}
                to="/dashboard"
              />
            </Switch>
          </div>,
          <UNDEFINED />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <CssBaseline />,
              <UNDEFINED />,
              <Switch>
                <Route
                  path="/dashboard"
                  render={[Function]}
                />
                <Route
                  path="/downloader"
                  render={[Function]}
                />
                <Route
                  path="/uploader"
                  render={[Function]}
                />
                <Route
                  path="/plugins"
                  render={[Function]}
                />
                <Route
                  path="/settings"
                  render={[Function]}
                />
                <Route
                  path="/account"
                  render={[Function]}
                />
                <Redirect
                  push={true}
                  to="/dashboard"
                />
              </Switch>,
            ],
            "className": "makeStyles-root-1",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(CtDrawer)",
                "type": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Route
                    path="/dashboard"
                    render={[Function]}
                  />,
                  <Route
                    path="/downloader"
                    render={[Function]}
                  />,
                  <Route
                    path="/uploader"
                    render={[Function]}
                  />,
                  <Route
                    path="/plugins"
                    render={[Function]}
                  />,
                  <Route
                    path="/settings"
                    render={[Function]}
                  />,
                  <Route
                    path="/account"
                    render={[Function]}
                  />,
                  <Redirect
                    push={true}
                    to="/dashboard"
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "path": "/dashboard",
                    "render": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "path": "/downloader",
                    "render": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "path": "/uploader",
                    "render": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "path": "/plugins",
                    "render": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "path": "/settings",
                    "render": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "path": "/account",
                    "render": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "push": true,
                    "to": "/dashboard",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(CtSnackbar)",
            "type": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
